clearpath_config/platform/wireless.py

```diff
@@ -0,0 +1,281 @@
+# Software License Agreement (BSD)
+#
+# @author    Chris Iverach-Brereton (civerachb@clearpathrobotics.com)
+# @copyright (c) 2025, Clearpath Robotics, Inc., All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# * Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# * Neither the name of Clearpath Robotics nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+from clearpath_config.common.types.config import BaseConfig
+from clearpath_config.common.utils.dictionary import flip_dict
+from clearpath_config.sensors.types.sensor import BaseSensor
+
+class BaseRouter():
+    """
+    Generic configuration parameters for a router we can install on the robot.
+
+    Must be sub-typed as appropriate.
+    """
+
+    IP_ADDRESS = 'ip_address'
+    LAUNCH_ENABLED = 'launch_enabled'
+
+    def __init__(
+        self,
+        ip_address: str = None,
+        launch_enabled: bool = True,
+    ):
+        self.ip_address = ip_address
+        self.launch_enabled = launch_enabled
+
+    def from_dict(self, d: dict):
+        self.ip_address = d.get(self.IP_ADDRESS, self.ip_address)
+        self.launch_enabled = d.get(self.LAUNCH_ENABLED, self.launch_enabled)
+
+    @property
+    def ip_address(self) -> str:
+        return self._ip_address
+
+    @ip_address.setter
+    def ip_address(self, value: str) -> None:
+        BaseSensor.assert_is_ipv4_address(value)
+        self._ip_address = value
+
+    @property
+    def launch_enabled(self) -> bool:
+        return self._launch_enabled
+
+    @launch_enabled.setter
+    def launch_enabled(self, value: bool):
+        self._launch_enabled = value
+
+
+class PeplinkRouter(BaseRouter):
```

Comment:
Does this work for any Peplink model, or just the one we support?

Reply:
It works with both models I have access to (the production model + the prototype installed in the AMP).

It's written to be compatible with Peplink's 8.5.3 API (but based on testing appears compatible with 8.5.2 as well), so it _should_ work with any model with compatible firmware.